tests/test_experimental/test_prediction_intervals/test_base.py

```diff
@@ -0,0 +1,196 @@
+        ),
+    ),
+)
+def test_pipelines_forecast_intervals(product_level_constant_hierarchical_ts, pipeline, expected_columns):
+    run_base_pipeline_compat_check(
+        ts=product_level_constant_hierarchical_ts, pipeline=pipeline, expected_columns=expected_columns
+    )
+
+
+@pytest.mark.parametrize(
+    "expected_columns",
+    ({"target", "target_lower", "target_upper"},),
+)
+@pytest.mark.parametrize(
+    "ensemble",
+    (
+        DirectEnsemble(pipelines=[get_naive_pipeline(horizon=1), get_naive_pipeline_with_transforms(horizon=2)]),
+        VotingEnsemble(pipelines=[get_naive_pipeline(horizon=1), get_naive_pipeline_with_transforms(horizon=1)]),
+        StackingEnsemble(pipelines=[get_naive_pipeline(horizon=1), get_naive_pipeline_with_transforms(horizon=1)]),
+    ),
+)
+def test_ensembles_forecast_intervals(example_tsds, ensemble, expected_columns):
+    run_base_pipeline_compat_check(ts=example_tsds, pipeline=ensemble, expected_columns=expected_columns)
+
+
+@pytest.mark.parametrize(
+    "pipeline",
+    (
+        Pipeline(model=CatBoostPerSegmentModel(), transforms=[DateFlagsTransform()]),
+        AutoRegressivePipeline(model=CatBoostPerSegmentModel(), transforms=[DateFlagsTransform()], horizon=1),
+        HierarchicalPipeline(
+            model=CatBoostPerSegmentModel(),
+            transforms=[DateFlagsTransform()],
+            horizon=1,
+            reconciliator=BottomUpReconciliator(target_level="market", source_level="product"),
+        ),
+    ),
+)
+@pytest.mark.parametrize(
+    "expected_params_to_tune",
+    (
+        {
+            "pipeline.model.learning_rate": FloatDistribution(low=1e-4, high=0.5, log=True),
+            "pipeline.model.depth": IntDistribution(low=1, high=11, step=1),
+            "pipeline.model.l2_leaf_reg": FloatDistribution(low=0.1, high=200.0, log=True),
+            "pipeline.model.random_strength": FloatDistribution(low=1e-05, high=10.0, log=True),
+            "pipeline.transforms.0.day_number_in_week": CategoricalDistribution([False, True]),
+            "pipeline.transforms.0.day_number_in_month": CategoricalDistribution([False, True]),
+            "pipeline.transforms.0.day_number_in_year": CategoricalDistribution([False, True]),
+            "pipeline.transforms.0.week_number_in_month": CategoricalDistribution([False, True]),
+            "pipeline.transforms.0.week_number_in_year": CategoricalDistribution([False, True]),
+            "pipeline.transforms.0.month_number_in_year": CategoricalDistribution([False, True]),
+            "pipeline.transforms.0.season_number": CategoricalDistribution([False, True]),
+            "pipeline.transforms.0.year_number": CategoricalDistribution([False, True]),
+            "pipeline.transforms.0.is_weekend": CategoricalDistribution([False, True]),
+            "width": FloatDistribution(low=-5.0, high=5.0),
+        },
+    ),
+)
+def test_params_to_tune(pipeline, expected_params_to_tune):
```

Comment:
This parametrization seems strange, I couldn't add pipeline that hasn't this `expected_params_to_tune`. 

May be we should it in a format like "pipeline, expected_params_to_tune". We could try some simple pipeline with sma, autoregressive with catboost, some simple hierarchical.

Reply:
Yes, we could, but here we mainly test that parameters are returned in the correct format for different pipeline types after wrapping with intervals. Basically, there is no difference in which model or transform we use. So models and transforms are fixed for test consistency.